### PR TITLE
[Core][Doc] add docs for out of disk prevention.

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

Update docs to reflect the out of disk prevention feature.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/objects/object-spilling.rst [hunk 1]
```diff
@@ -68,6 +68,26 @@ If you are using an HDD, it is recommended that you specify a large buffer size
...
 
+To prevent running out of disk space, object spilling will throw ``OutOfDiskError`` if the local disk utilization exceeds the predefined threshold.
+If multiple physical devices are used, any physical device's over-usage will trigger the ``OutOfDiskError``.
+The default threshold is 0.95 (95%). You can adjust the threshold by setting ``local_fs_capacity_threshold``, or set it to 1 to disable the protection.
+
+.. code-block:: python
+
+    ray.init(
+        _system_config={
+            "local_fs_capacity_threshold": 0.99, # Allow spilling until local disk is 99% utilized.
```
Comment (jjyao) [inline doc/source/ray-core/objects/object-spilling.rst:79]: Maybe worth mentioning that this doesn't apply to remote storage?

**Review by ericl — APPROVED** [review]

File: doc/source/ray-core/objects/object-spilling.rst [hunk 2]
```diff
@@ -68,6 +68,28 @@ If you are using an HDD, it is recommended that you specify a large buffer size
         },
     )
 
+To prevent running out of disk space, local object spilling will throw ``OutOfDiskError`` if the disk utilization exceeds the predefined threshold.
+If multiple physical devices are used, any physical device's over-usage will trigger the ``OutOfDiskError``.
```
Comment (rkooo567) [inline doc/source/ray-core/objects/object-spilling.rst:72]: Should we mention it is for /tmp mounted devices?

**Review by rkooo567 — APPROVED** [review]
Can you update the PR description? (just for the recored)

File: doc/source/ray-core/objects/object-spilling.rst (hunk 2 above)
Comment (scv119) [reply doc/source/ray-core/objects/object-spilling.rst:72]: hmm this is not true right; we monitor the disk where the spilling is configured

**Pull request merged**